### PR TITLE
Upgrade to nom 6

### PR DESCRIPTION
**Pull request by nickelc**
The most of changes are related to the switch from `Fn` to `FnMut`.

### CODE REVIEW

File: src/ber/parser.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use nom::number::streaming::be_u8;
 use nom::*;
 use rusticata_macros::{custom_check, parse_hex_to_u64};
 use std::borrow::Cow;
-use std::convert::TryFrom;
+use std::convert::{Into, TryFrom};
```
Comment (nickelc) [inline src/ber/parser.rs:11]: `Into` must be imported here because of `nom::*` and the new `nom::Into` type.

**Comment by chifflier** [comment]
Hi,
Thanks for your contribution. I had already created a local branch here for the nom 6 upgrade, and have to compare it to your changes before deciding which one to merge. At the first view, I see similar changes with minor differences.
I see also that MSRV will have to be changed, nom 6 requires rust 1.44+

**Comment by nickelc** [comment]
The MSRV could be way lower without the bitvec dependencies.

**Comment by chifflier** [comment]
> The MSRV could be way lower without the bitvec dependencies.

`bitvec` could be interesting, especially for BITSTRING (currently parsed as slice)
That said, it brings a lot of dependencies. I'll run some tests, maybe we can use only `std` for `nom`.

**Comment by nickelc** [comment]
> > The MSRV could be way lower without the bitvec dependencies.
> 
> `bitvec` could be interesting, especially for BITSTRING (currently parsed as slice)
> That said, it brings a lot of dependencies. I'll run some tests, maybe we can use only `std` for `nom`.

nom/std still pulls bitvec

> > The MSRV could be way lower without the bitvec dependencies.
> 
> `bitvec` could be interesting, especially for BITSTRING (currently parsed as slice)
> That said, it brings a lot of dependencies. I'll run some tests, maybe we can use only `std` for `nom`.

i tried that with `rusticata-macros` but it still pulls bitvec.

**Comment by chifflier** [comment]
> i tried that with `rusticata-macros` but it still pulls bitvec.

Indeed. I guess we have to keep it for now

**Comment by chifflier** [comment]
Merged (with some additional commits), thanks!

**Pull request merged**